### PR TITLE
KOGITO-1547: Change online modeler to have a link to download the Hub

### PR DESCRIPTION
**Pull request by ljmotta**
## Task
https://issues.redhat.com/browse/KOGITO-1547

## Description
- Removed the links to GitHub and VS Code extensions.
- Added the link to the Get Business Modeler Hub Preview modal.
- Modal with description of all Kogito tools and the three operating systems options for download: Linux, macOS and Windows.
- Modal with a successful message and option to download using a link.

## Demo
kogito-1547

## Considerations
Regardless the successful modal, it wasn't made following the prototype (https://marvelapp.com/223hdgb6/screen/67389180). The download starts instantly, that way the spinner even wasn't going to show up.

### CODE REVIEW

**Comment by tiagobento** [comment]
Thanks @ljmotta! I'll update this PR with the correct links to the Hub artifacts after #91 is merged.

**Review by tiagobento — APPROVED** [review]

**Review by paulovmr — APPROVED** [review]

File: packages/online-editor/src/home/DownloadHubModal.tsx [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+enum ModalState {
+  SELECT_OS,
+  DOWNLOADED,
+  CLOSE
+}
+
+const availableOperatingSystems = new Map<OperatingSystem, string>([
+  [OperatingSystem.LINUX, "Linux"],
+  [OperatingSystem.MACOS, "macOS"],
+  [OperatingSystem.WINDOWS, "Windows"]
```
Comment (jomarko) [inline packages/online-editor/src/home/DownloadHubModal.tsx:41]: Please why we use other values than in `utils.ts`?

```
export enum OperatingSystem {
  MACOS = "MACOS",
  WINDOWS = "WINDOWS",
  LINUX = "LINUX"
}
```

**Review by jomarko — COMMENTED** [review]
### Code review

**Review by jomarko — COMMENTED** [review]
On my machine, once I built PR and started online editor according `kogito-tooling/README` it tries to download from url `https://github.com/kiegroup/kogito-tooling/releases/download/0.2.4/business_modeler_hub_preview_linux_0.2.4.zip` which is not existing, is that expected?

**Comment by jomarko** [comment]
The focused element in dialogue after opening is top right `cross`, so basically if user press `Enter` the dialogue is closed? Isn't expected the primary download button has focus or the dropdown?
Screenshot from 2020-04-02 16-31-57

File: packages/online-editor/src/home/DownloadHubModal.tsx (hunk 1 above)
Comment (ljmotta) [reply packages/online-editor/src/home/DownloadHubModal.tsx:41]: The values on `DownloadHubModal.tsx` are displayed on the modal for the final user, and the values in `utils.ts` are the ones used on the editor context.

**Comment by paulovmr** [comment]
@jomarko Locally the link will only work after we make the first release for the hub app, and if you have the latest tags fetched in your local copy.

**Comment by jomarko** [comment]
@paulovmr ok, I tried to search for something downloadable but even `https://github.com/kiegroup/kogito-tooling/releases/download` gives me `404`

**Comment by paulovmr** [comment]
The parent paths in the link don't list files, so you would have to use the link directly to the asset you want, e.g.: [https://github.com/kiegroup/kogito-tooling/releases/download/0.2.10/chrome_extension_kogito_kie_editors_0.2.10.zip](https://github.com/kiegroup/kogito-tooling/releases/download/0.2.10/chrome_extension_kogito_kie_editors_0.2.10.zip)

**Comment by tiagobento** [comment]
@jomarko Thanks for testing this PR so quickly. I think that the focus moving to the close button is something that is done by design by PatternFly. @srambach Can you please confirm if that's the case?

**Comment by jomarko** [comment]
https://github.com/kiegroup/kogito-tooling/pull/97#issuecomment-607884612
@paulovmr so my question is, can I approve this before hub is released? How can I double check otherwise the download works?

**Comment by kywalker-rh** [comment]
@ljmotta Its hard to tell from the video, is the "Get Business Modeler Hub" link a link button, or a secondary button? It looks like there is a blue line around it at one point in the video. For reference my intention was that it would be just text, not a visual button.

**Comment by paulovmr** [comment]
@jomarko to test the download itself you can do e.g. `yarn start --DOWNLOAD_HUB_linuxUrl=https://github.com/kiegroup/kogito-tooling/releases/download/0.2.10/chrome_extension_kogito_kie_editors_0.2.10.zip` to run the online editor.

**Comment by tiagobento** [comment]
@ljmotta @jomarko btw just tested this PR on macOS and Windows, and the combo box is correctly pre-selected.

**Comment by ljmotta** [comment]
@kywalker-rh  Hi there! I used a Button component with the variant "plain". I'm going to make the change for just text, sorry about that.

**Comment by kywalker-rh** [comment]
@ljmotta No problem at all, thanks for updating it!

**Comment by ljmotta** [comment]
@kywalker-rh a demo of the fix:

ezgif com-video-to-gif

If the PR has any other fix that is needed, let me know

**Comment by kywalker-rh** [comment]
@ljmotta Perfect, thanks so much! I did just notice the button in the downloading window says close instead of cancel. My thought was that would be a way to cancel the download if it was taking too long, or there was an issue with it. Is that possible?

**Comment by jomarko** [comment]
@paulovmr thank you, it helped a lot, the positive scenario works nice. Hoever last question about negative scenario, I mean tried to download for platform I didn't specified existing path, it shows me for short time the 'next step' dialogue. Is it expected/acceptable? 

[video](https://issues.redhat.com/secure/attachment/12469232/unset-url.webm) attached to jira.

**Comment by ljmotta** [comment]
@kywalker-rh To have a progress check or a control over the download, a server would be necessary, and we don't have that yet. Right now, it's not possible to cancel the download with a button. The spinner was removed because of the same reason. To be more concise with my changes, I changed the modal title as well. I hope that is OK!

**Comment by paulovmr** [comment]
@jomarko IMO I think it is acceptable, because it will only happen during a github instability. Also there isn't any side effect.

**Comment by tiagobento** [comment]
@kywalker-rh Note that once the downloads starts, you can use your browser's download manager/features to cancel/pause etc.

**Review by jomarko — APPROVED** [review]
thank you

**Comment by srambach** [comment]
> @jomarko Thanks for testing this PR so quickly. I think that the focus moving to the close button is something that is done by design by PatternFly. @srambach Can you please confirm if that's the case?

After checking with the PF team, focus first to the close button is the current expected behavior. However, providing for the application to specify where the focus goes has been discussed. I'm putting in an issue on it. For now, I guess I'd say leave it as is.

**Review by ederign — APPROVED** [review]

**Comment by kywalker-rh** [comment]
@ljmotta Thanks for letting me know, that makes perfect sense!

**Pull request merged**